components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dao/impl/IdentityUserMetadataMgtDAOImpl.java

```diff
@@ -0,0 +1,508 @@
+     * @return              boolean value of the execution.
+     * @throws SQLException SQL Exception.
+     */
+    private boolean updateExistingMetadata(Connection connection, String value, String tenantId, String username,
+                                           String claimURI) throws SQLException {
+
+        String sqlStmt = IdentityRecoveryConstants.SQLQueries.UPDATE_USER_METADATA;
+        PreparedStatement prepStmt = connection.prepareStatement(sqlStmt);
+        prepStmt.setString(1, value);
+        prepStmt.setString(2, tenantId);
+        prepStmt.setString(3, username);
+        prepStmt.setString(4, claimURI);
+        int result = prepStmt.executeUpdate();
+        return result > 0;
+    }
+
+    /**
+     * Insert new user metadata.
+     *
+     * @param value         Value of the claim.
+     * @param tenantId      Tenant Id.
+     * @param username      Username.
+     * @param claimURI      Claim URI.
+     * @return              boolean value of the execution.
+     * @throws SQLException SQL Exception.
+     */
+    private boolean insertNewMetadata(Connection connection, String value, String tenantId, String username,
+                                      String claimURI) throws SQLException {
+
+        String sqlStmt = IdentityRecoveryConstants.SQLQueries.INSERT_USER_METADATA;
+        PreparedStatement prepStmt = connection.prepareStatement(sqlStmt);
+        prepStmt.setString(1, tenantId);
+        prepStmt.setString(2, username);
+        prepStmt.setString(3, claimURI);
+        prepStmt.setString(4, value);
+        int result = prepStmt.executeUpdate();
+        return result > 0;
+    }
+
+    /**
+     * Check weather the given user store has enabled the property "StoreIdentityClaims" to store identity claims
+     * in the user store.
+     *
+     * @param userStoreManager  User Store manager.
+     * @return                  Whether identity claims are stored in user store or not.
+     */
+    private boolean isStoreIdentityClaimsInUserStoreEnabled(UserStoreManager userStoreManager) {
+
+        return Boolean.parseBoolean(userStoreManager.getRealmConfiguration().
+                getUserStoreProperty(IdentityRecoveryConstants.STORE_IDENTITY_CLAIMS));
+    }
+
+    /**
+     * Fetch relevant SQL statement from user store configuration.
+     *
+     * @param userStoreManager          User Store manager.
+     * @param sqlStatementKey           Property name of the SQL statement.
+     * @return                          SQL statement string.
+     * @throws IdentityEventException   Error while fetching the SQL statement.
+     */
+    private String fetchSQLStatementFromUserStoreConfig(UserStoreManager userStoreManager, String sqlStatementKey)
+            throws IdentityEventException {
+
+        String sqlStmt = userStoreManager.getRealmConfiguration().getUserStoreProperty(sqlStatementKey);
+        if (StringUtils.isNotBlank(sqlStmt)) {
+            return sqlStmt;
+        } else {
```

Comment:
else is redundant right? we can throw the exception without the else block.

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/697/commits/dd3419fd69cc40e847e40ed9dfb29c867a3f0018